homeassistant/components/stream/__init__.py

```diff
@@ -0,0 +1,156 @@
+"""
+Provide functionality to stream video source.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/stream/
+"""
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.auth.util import generate_secret
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.loader import bind_hass
+
+from .const import (
+    DOMAIN, CONF_KEEPALIVE, ATTR_STREAMS, ATTR_ENDPOINTS)
+from .worker import stream_worker
+from .hls import async_setup_hls
+
+REQUIREMENTS = ['av==6.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_KEEPALIVE, default=False): cv.boolean,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+# pylint: disable=dangerous-default-value
+@bind_hass
+async def async_request_stream(hass, stream_source, options={}, preload=False):
+    """Set up stream with token."""
+    if DOMAIN not in hass.config.components:
+        raise HomeAssistantError("Stream is not configured.")
```

Comment:
```suggestion
        raise HomeAssistantError("Stream component is not set up.")
```

Reply:
I don't know if I agree here.  If the component failed to setup properly, do we want to do this check in `n` places, or just here?  If this method is the de-facto way to request a stream, I don't see an issue with it doing it's own error checking.